### ISSUE TITLE
uncaught exception in CreateSnapshot (NFS)

### ISSUE THREAD

**Issue by tgross** [democratic-csi/democratic-csi#166]
I encountered an uncaught exception while trying to snapshot a NFS volume with `docker.io/democraticcsi/democratic-csi:v1.5.4`. The snapshot directory is created on the remote host but the contents were not rsync'd. It's possible (even likely! 😀 ) that there's a NFS configuration error here or something we're doing wrong on the Nomad side. But I imagine y'all would want to know about a crash even in that case.

My configuration:

```yaml
driver: nfs-client
instance_id:
nfs:
  shareHost: 192.168.56.69
  shareBasePath: "/var/nfs/general"
  controllerBasePath: "/storage"
  dirPermissionsMode: "0777"
  dirPermissionsUser: root
  dirPermissionsGroup: root
```

Logs from the plugin:

> {"level":"info","message":"new request - driver: ControllerNfsClientDriver method: CreateSnapshot call: {\"_events\":{},\"_eventsCount\":1,\"call\":{},\"cancelled\":false,\"metadata\":{\"_internal_repr\":{\"user-agent\":[\"grpc-go/1.44.1-dev\"]},\"flags\":0},\"request\":{\"secrets\":\"redacted\",\"parameters\":{},\"source_volume_id\":\"csi-volume-nfs3\",\"name\":\"test-snapshot-3\"}}","service":"democratic-csi"}
> {"level":"debug","message":"operation lock keys: [\"create_snapshot_name_test-snapshot-3\",\"volume_id_csi-volume-nfs3\"]","service":"democratic-csi"}
> {"level":"verbose","message":"requested snapshot name: test-snapshot-3","service":"democratic-csi"}
> {"level":"verbose","message":"cleansed snapshot name: test-snapshot-3","service":"democratic-csi"}
executing command: mkdir -p /storage/s/csi-volume-nfs3-test-snapshot-3
executing command: rsync -a /storage/v/csi-volume-nfs3/ /storage/s/csi-volume-nfs3-test-snapshot-3/
{"date":"Tue Mar 22 2022 20:02:07 GMT+0000 (Coordinated Universal Time)","error":{"code":"ENOENT","errno":"ENOENT","path":"rsync","spawnargs":["-a","/storage/v/csi-volume-nfs3/","/storage/s/csi-volume-nfs3-test-snapshot-3/"],"syscall":"spawn rsync"},"exception":true,"level":"error","message":"uncaughtException: spawn rsync ENOENT\nError: spawn rsync ENOENT\n    at Process.ChildProcess._handle.onexit (internal/child_process.js:268:19)\n    at onErrorNT (internal/child_process.js:470:16)\n    at processTicksAndRejections (internal/process/task_queues.js:84:21)","os":{"loadavg":[0.01,0.05,0],"uptime":6902},"process":{"argv":["/usr/local/bin/node","/home/csi/app/bin/democratic-csi","--csi-version=1.5.0","--csi-name=org.democratic-csi.nfs","--driver-config-file=/local/driver-config-file.yaml","--log-level=debug","--csi-mode=controller","--server-socket=unix:///csi/csi.sock"],"cwd":"/home/csi/app","execPath":"/usr/local/bin/node","gid":0,"memoryUsage":{"arrayBuffers":78716,"external":18297196,"heapTotal":18984960,"heapUsed":17048048,"rss":59924480},"pid":1,"uid":0,"version":"v12.22.6"},"service":"democratic-csi","stack":"Error: spawn rsync ENOENT\n    at Process.ChildProcess._handle.onexit (internal/child_process.js:268:19)\n    at onErrorNT (internal/child_process.js:470:16)\n    at processTicksAndRejections (internal/process/task_queues.js:84:21)","trace":[{"column":19,"file":"internal/child_process.js","function":"Process.ChildProcess._handle.onexit","line":268,"method":"onexit","native":false},{"column":16,"file":"internal/child_process.js","function":"onErrorNT","line":470,"method":null,"native":false},{"column":21,"file":"internal/process/task_queues.js","function":"processTicksAndRejections","line":84,"method":null,"native":false}]}
running server shutdown, exit code: Error: spawn rsync ENOENT

**Comment by tgross** [comment]
Oh... if I exec into the plugin container and try running the `rsync` command from there, I get the following. The `rsync` binary doesn't appear to be anywhere on the PATH.

```
$ nomad alloc exec ff79 /bin/bash
root@fb921baebc83:/home/csi/app# rsync -a /storage/v/csi-volume-nfs3/ /storage/s/csi-volume-nfs3-test-snapshot-3/
bash: rsync: command not found
```

**Comment by travisghansen** [comment]
Ah! I better fix that lol. I’ll get a new image built with that in there.

**Comment by travisghansen** [comment]
Fix here: https://github.com/democratic-csi/democratic-csi/commit/810025150ca60c9e44cff0f519dcb87b4b91c791 it'll take a bit build/test the images and get everything tagged but I'll shout out when `v1.6.1` images are built.

**Comment by travisghansen** [comment]
OK, included in `next`, `latest` and `v1.6.1` now. Not sure if `rsync` was always missing in the images or if it disappeared during the update from `buster` to `bullseye` but thanks for the report.

Try it out and let me know if it works as desired/designed.

**Comment by tgross** [comment]
I can confirm it's working now with 1.6.1:

```
$ nomad volume snapshot create csi-volume-nfs4 test-snapshot-4
Snapshot ID                      Volume ID     Size     Create Time                Ready?
csi-volume-nfs4-test-snapshot-4  csi-volume-n  8.0 KiB  2022-03-25T11:07:50-04:00  true
```

Thanks @travisghansen!
